src/main/java/org/testng/internal/ConstructorOrMethod.java

```diff
@@ -8,7 +10,7 @@
  *
  * @author Cedric Beust <cedric@beust.com>
  */
-public class ConstructorOrMethod {
+public class ConstructorOrMethod implements IFactoryMethod {
```

Comment:
A `IFactoryMethod` is a `ConstructorOrMethod` but a `ConstructorOrMethod` is not a `IFactoryMethod`

Reply:
Yes and No. In one aspect, what you say is correct, but if you look at an implementation perspective, then only a constructor or a method can be a factory method. There are two ways of looking at it. The easiest way for me would have been to directly expose `ConstructorOrMethod` and be done with it. But I believe you had concerns around an internal class being exposed to outside users. 

This was the only way I could think of, to fix this.

Reply:
And why not `IFactoryMethod#getConstructorOrMethod` like `ITestNGMethod`?

Reply:
I think we might as well just get rid of the `IFactoryMethod` and replace it with `IConstructorOrMethod` in that case, because if I were to implement `IFactoryMethod#getConstructorOrMethod` then I would not know what would the `IFactoryMethod` interface contain. It was introduced only to prevent `ConstructorOrMethod` internal class from being exposed outside. The other option is that we just expose it just as `org.testng.ITestNGMethod#getConstructorOrMethod`

Reply:
It looks we have 2 different objectives for `IFactoryMethod` :) I hope my previous comments will be clear enough to explain my vision.